examples/tn_simulation.py

```diff
@@ -0,0 +1,233 @@
+    for j in range(0, iterations):
+        for i in range(1, n):
+            circuit.cnot(i, 0)
+            circuit.rz(0, gamma[j])
+            circuit.cnot(i, 0)
+        for i in range(0, n):
+            circuit.rx(i, beta[j])
+
+
+# %%
+# Let's look at how the quantum circuit is going to be built.
+
+n = 5  # This will result in a graph that would be too large for statevector simulation.
+iterations = 2  # Define the number of iterations in the quantum circuit.
+gamma = 2 * np.pi * np.random.rand(iterations)
+beta = 2 * np.pi * np.random.rand(iterations)
+# Define the circuit.
+circuit = Circuit(n)
+ansatz(circuit, n, gamma, beta, iterations)
+
+# Transpile Circuit into pattern as it is needed for creating the TN.
+pattern = circuit.transpile(opt=True).pattern
+# Optimizing according to standardization algorithm of graphix.
+pattern.standardize()
+pattern.shift_signals()
+
+# %%
+# Print some properties of the graph.
+
+nodes, edges = pattern.get_graph()
+print(f"Number of nodes: {len(nodes)}")
+print(f"Number of edges: {len(edges)}")
+
+# %%
+# Optimizing by performing Pauli measurements in the pattern using efficient stabilizer simulator.
+
+pattern.perform_pauli_measurements(use_rustworkx=True)
+
+# %%
+# Simulate using the TN backend of graphix, which will return an MBQCTensorNet object.
+# The graph_prep argument is optional,
+# but with 'parallel' the TensorNetworkBackend will prepeare the graph state faster.
+
+mbqc_tn = pattern.simulate_pattern(backend="tensornetwork", graph_prep="parallel")
+sv = mbqc_tn.to_statevector().flatten()
+print("Statevector after the simulation:", sv)
+
+# %%
+# Let's explore what is really happening, how the TN is being constructed.
+# The tensor network is created using the graph structure, so from the list of nodes as well as the edges.
+# The graph must be preprocessed before the constraction of the TN itself.
+# For every node a list is created, which stores the index labels for each dimension for that given node.
+# Its length will be one larger than the number of edges of the node (for more information look for properties of MPS).
+# Additionally, the type of edges are also stored in a binary valued list for each node.
+# These are used to construct the tensor itself later.
+# In the following, the TN construction can finally take place.
+# From each node in the graph a tensor is constructed, which has a dimension that is exactly one larger than its neighbour count.
+# The tensor is described using two outer products for which the list used from above, that describes the edges for every node.
+# For additional information on TN construction please refer to: https://journals.aps.org/pra/abstract/10.1103/PhysRevA.76.052315
+
+# %%
+# Let's also plot the resulting tensor network.
+
+fig, ax = plt.subplots(figsize=(13, 10))
+color = ["Z", "M", "X", "ancilla"]
+mbqc_tn.draw(
+    ax=ax,
+    color=color,
+    show_tags=False,
+    show_inds=False,
+    layout="kamada_kawai",
```

Comment:
can you remove this argument `layout` as this is not available in quimb ver 1.4.2 (minimum requiremnt for graphix)? hope it still render ok?

Reply:
I removed the argument and looked for ways to use the recommended function, but ended up not using it after all. However, I changed the coloring of the graph to be more in sync with the description.

Reply:
@Barnald I see, I wonder why it did not work? maybe I can help. 
the positioning of `get_pos_from_flow` (or gflow ver) will be easier to digest in comparison with the `draw_graph` of pattern, so it would be great if this works.

Reply:
Yeah I completely agree that it would be great to get it working. The issue is that the ``get_pos_from_flow`` function expects a flow, but it have to be constructed somehow from the ``MBQCTensorNet``. I tried to construct the flow before, and will take another look at it

Reply:
I managed to parse enough information to be able to use ``get_pos_from_gflow``, and I think it should be working